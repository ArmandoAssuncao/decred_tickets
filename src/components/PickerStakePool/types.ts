import { ITicketState } from 'store/ticket/types'

export interface IProps {
    value?: ITicketState['stakePool']
    onChange?: (value: ITicketState['stakePool']) => void
}

export interface IState {
}
