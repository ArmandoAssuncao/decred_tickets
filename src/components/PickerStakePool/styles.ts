import { ViewStyle } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'

export default EStyleSheet.create({
    picker: {
        backgroundColor: '$secondColor',
        borderRadius: 10,
        width: undefined,
    } as ViewStyle,
})
