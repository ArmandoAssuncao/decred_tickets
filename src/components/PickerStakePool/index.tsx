import React from 'react'
import { Icon, Picker } from 'native-base'

import { IProps, IState } from './types'
import styles from './styles'

import { ITicketState } from 'store/ticket/types'
import i18n from 'config/i18n'

export default class PickerStakePool extends React.Component<IProps, IState> {
    public static defaultProps: Partial<IProps> = {
        onChange: null,
    }

    render(): JSX.Element {
        const { onChange, value } = this.props

        const stakePools: Array<ITicketState['stakePool']> = [
            'decredBrasil',
        ]

        const textType = 'StakePool'

        return (
            <Picker
                mode="dropdown"
                iosHeader={i18n.t('components.picker.header', {type: textType})}
                headerBackButtonText={i18n.t('components.picker.backBtn')}
                iosIcon={<Icon name="ios-arrow-down-outline" color="black" />}
                placeholder={i18n.t('components.picker.placeholder', {type: textType})}
                style={styles.picker}
                selectedValue={value}
                onValueChange={onChange}
            >
                <Picker.Item
                    label={i18n.t('components.picker.header', {type: textType}).toUpperCase()}
                    value={null}
                />
                {
                    stakePools.map((stake, i) => (
                        <Picker.Item
                            key={i.toString()}
                            label={i18n.t(`decred.stakePools.${stake}`)}
                            value={stake}
                        />
                    ))
                }
            </Picker>
        )
    }
}
