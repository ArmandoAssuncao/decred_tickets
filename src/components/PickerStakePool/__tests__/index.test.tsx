import React from 'react'
import 'react-native'
import { shallow } from 'enzyme'

import PickerStakePool from '../index'

describe('PickerStakePool', () => {
    const wrapper = shallow(<PickerStakePool value={'decredBrasil'} onChange={jest.fn()} />)

    it('match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('has 1 Styled(PickerNB)', () => {
        expect(wrapper.find('Styled(PickerNB)')).toHaveLength(1)
    })

    it('has 2 Item', () => {
        expect(wrapper.find('Item')).toHaveLength(2)
    })

    it('has Item whit prop value equal null', () => {
        expect(wrapper.find('Item').at(0).prop('value')).toEqual(null)
    })

    it('has Item whit prop value equal "decredBrasil"', () => {
        expect(wrapper.find('Item').at(1).prop('value')).toEqual('decredBrasil')
    })

    describe('actions', () => {
        it('call onValueChange when pressed', () => {
            wrapper.props().onValueChange()
            expect(wrapper.prop('onValueChange')).toBeCalled()
        })
    })
})
