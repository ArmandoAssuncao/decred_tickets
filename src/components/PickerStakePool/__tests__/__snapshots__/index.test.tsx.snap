// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerStakePool match snapshot 1`] = `
<Styled(PickerNB)
  headerBackButtonText="Back"
  iosHeader="Select StakePool"
  iosIcon={
    <Styled(Icon)
      color="black"
      name="ios-arrow-down-outline"
    />
  }
  mode="dropdown"
  onValueChange={[MockFunction]}
  placeholder="Select StakePool"
  selectedValue="decredBrasil"
  style={
    Object {
      "backgroundColor": "#F3F6F6",
      "borderRadius": 10,
      "width": undefined,
    }
  }
>
  <Item
    label="SELECT STAKEPOOL"
    value={null}
  />
  <Item
    key="0"
    label="stake.decredbrasil.com"
    value="decredBrasil"
  />
</Styled(PickerNB)>
`;
